include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp

```diff
@@ -0,0 +1,162 @@
+
+#include <OpenSpaceToolkit/Core/Types/Shared.hpp>
+#include <OpenSpaceToolkit/Core/Types/Size.hpp>
+#include <OpenSpaceToolkit/Core/Types/String.hpp>
+
+#include <OpenSpaceToolkit/Mathematics/Objects/Vector.hpp>
+
+#include <OpenSpaceToolkit/Physics/Coordinate/Frame.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+
+using ostk::core::types::Shared;
+using ostk::core::types::Size;
+using ostk::core::types::String;
+
+using ostk::math::obj::VectorXd;
+
+using ostk::physics::coord::Frame;
+using ostk::physics::time::Instant;
+
+class CoordinatesBroker;
+
+/// @brief  State coordinates subset. It contains information related to a particular group of coordinates. It does not
+/// hold coordinates value information.
+
+class CoordinatesSubset
+{
+   public:
+    /// @brief              Constructor
+    ///
+    /// @param              [in] aName A name
+    /// @param              [in] aSize A size (i.e. dimension)
+
+    CoordinatesSubset(const String& aName, const Size& aSize);
+
+    /// @brief              Equals to operator
+    ///
+    /// @param              [in] aCoordinatesSubset The coordinate subset to compare it to
+    ///
+    /// @return             True if both instances refer to the same subset and have the same size (i.e. dimension)
+
+    bool operator==(const CoordinatesSubset& aCoordinatesSubset) const;
+
+    /// @brief              Not equals to operator
+    ///
+    /// @param              [in] aCoordinatesSubset The coordinate subset to compare it to
+    ///
+    /// @return             False if both instances refer to the same subset and have the same size (i.e. dimension)
+
+    bool operator!=(const CoordinatesSubset& aCoordinatesSubset) const;
+
+    /// @brief              Returns the unique identifier of the instance
+    ///
+    /// @return             The unique identifier of the instance
+
+    String getId() const;
+
+    /// @brief              Returns the name of the instance
+    ///
+    /// @return             The name of the instance
+
+    String getName() const;
+
+    /// @brief              Returns the size (i.e. dimension) of the instance
+    ///
+    /// @return             The size (i.e. dimension) of the instance
+
+    Size getSize() const;
+
+    /// @brief              Adds two coordinates subsets
+    ///
+    /// @param              [in] anInstant the instant associated to the coordinates
+    /// @param              [in] allCoordinates_1 first set of all coordinates
+    /// @param              [in] allCoordinates_2 second set of all coordinates
+    /// @param              [in] aFrame the reference frame in which the coordinates are resolved
+    /// @param              [in] aCoordinatesBroker a coordinates broker
+    ///
+    /// @return             The resulting coordinates subset value (subset_1 + subset_2)
+
+    VectorXd add(
+        const Instant& anInstant,
+        const VectorXd& allCoordinates_1,
```

Comment:
```suggestion
        const VectorXd& allCoordinates_1,
```
I'd use `afullCoordinates` and `anotherFullCoordinates` to align